### DISCUSSION TITLE
Trimgalore/cutadapt not creating FastQC html reports

### DISCUSSION THREAD

**Discussion by yaaminiv** [RobertsLab/resources#1104] [Q&A] [answered]
I am running [this mox script](https://github.com/RobertsLab/project-gigas-oa-meth/blob/master/code/02-trimgalore.sh) on Manchester *C. gigas* gonad WGBS data. The files trimmed, but none of the fastqc HTML reports were generated! I ran [a similar script](https://github.com/RobertsLab/project-oyster-oa/blob/master/code/Haws/01-trimgalore.sh) on the Hawaii data and got HMTL reports, so I'm not sure why that's not happening this time. How can I get these reports from my TrimGalore/cutadapt code (besides running FastQC separately)?

[Hawaii run slurm](https://gannet.fish.washington.edu/spartina/project-oyster-oa/Haws/trimmed-data/slurm-740304.out)
[Manchester run slurm](https://gannet.fish.washington.edu/spartina/project-gigas-oa-meth/output/trimgalore/01-abundant-adapters/slurm-1076838.out)

*edit: added correct slurm links*

**Comment by kubu4** [comment, marked as answer]
This error message is in your Hawaii slurm:

```
  >>> Now running FastQC on the validated data zr3616_8_R1_val_1.fq.gz<<<

Can't exec "fastqc": No such file or directory at /gscratch/srlab/programs/TrimGalore-0.6.6/trim_galore line 1525, <IN2> line 5536487816.

  >>> Now running FastQC on the validated data zr3616_8_R2_val_2.fq.gz<<<

Can't exec "fastqc": No such file or directory at /gscratch/srlab/programs/TrimGalore-0.6.6/trim_galore line 1535, <IN2> line 5536487816.
Deleting both intermediate output files zr3616_8_R1_trimmed.fq.gz and zr3616_8_R2_trimmed.fq.gz
```

I don't have time ATM to investigate the error, but the other option is to run FastQC independently, like so. It will run on all files matching this pattern (`*.fq.gz`), so you don't need to run FastQC one-by-one on each file:

```bash
# Set CPU threads to use
threads=28

# Populate array with FastQ files
fastq_array=(*.fq.gz)

# Pass array contents to new variable
fastqc_list=$(echo "${fastq_array[*]}")

# Run FastQC
# NOTE: Do NOT quote ${fastqc_list}
fastqc \
--threads ${threads} \
--outdir ${output_dir} \
${fastqc_list}
```

**Comment by kubu4** [comment]
Per this [TrimGalore Issue](https://github.com/FelixKrueger/TrimGalore/issues/112), they say the error message indicates that the command couldn't find FastQC. 

Is there some possibility that FastQC isn't in your system $PATH any more?

**Reply by yaaminiv** [reply]
Yup, I don't think it's in my path anymore. I tried running the code you posted above without specifying the fastqc directory and my script failed. I'll add it back to my path!

**Comment by sr320** [comment]
Or.....    always use the full path :)

On Sun, Feb 7, 2021 at 4:51 PM Yaamini Venkataraman <
notifications@github.com> wrote:

> Yup, I don't think it's in my path anymore. I tried running the code you
> posted above without specifying the fastqc directory and my script failed.
> I'll add it back to my path!
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/RobertsLab/resources/discussions/1104#discussioncomment-348008>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABB4PN5AFXDTCJWVL23O3VTS54YOXANCNFSM4XDSX5OQ>
> .
>
-- 

Steven B. Roberts, Associate Professor

Associate Director - Graduate Program Coordinator

School of Aquatic and Fishery Sciences

University of Washington

Fisheries Teaching and Research (FTR) Building - Office 232

1140 NE Boat Street - Seattle, WA 98105

robertslab.info - sr320@uw.edu - @sr320

vm:206.866.5141 - cell:360.362.3626

**Comment by kubu4** [comment]
Here's a "pro" tip when running SLURM scripts on Mox (or, any computer, really). Add this to the bottom of your SLURM script:

```shell
# Document programs in PATH
{
date
echo ""
echo "System PATH for $SLURM_JOB_ID"
echo ""
printf "%0.s-" {1..10}
echo "${PATH}" | tr : \\n
} >> system_path.log

echo "Finished logging system PATH"
```

This creates a newline-delimited log of all the items listed in your system `$PATH` for your SLURM job. It's a nice record to have, especially in instances like this where something appears broken _and_ is possibly related to system `$PATH` issues.

I'll add this somewhere to our Handbook...
